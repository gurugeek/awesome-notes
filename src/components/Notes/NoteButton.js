import React from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import Icon from 'react-native-vector-icons/Ionicons';

import { ICON_TYPES } from '../../constants/IconTypes';
import  { NotesStyle } from '../../styles';

const NoteButton = ({ type }) => {
    return (
        <View style={NotesStyle.noteButton.wrapper}>
            <Icon name={ICON_TYPES[type].name} size={30} color={ICON_TYPES[type].color} />
        </View>
    );
};

NoteButton.propTypes = {
    type: PropTypes.string.isRequired
};

export default NoteButton;