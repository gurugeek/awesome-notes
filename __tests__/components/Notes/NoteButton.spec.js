import React from 'react';
import { View } from 'react-native';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Icon from 'react-native-vector-icons/Ionicons';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import NoteButton from '../../../src/components/Notes/NoteButton';

describe('<NoteButton /> Component', () => {
    it('NoteButton should match with snapshot', () => {
       const tree = renderer.create(
           <NoteButton type='plus' />
       ) ;
        const json = tree.toJSON();
        expect(json).toMatchSnapshot();
    });

    const noteButtonShallow = shallow(<NoteButton type='plus' />);

    it('NoteButton should exists', () => {
        expect(noteButtonShallow.length).toEqual(1);
    });

    it('NoteButton should has one View', () => {
        expect(noteButtonShallow.find(View)).toHaveLength(1);
    });

    it('NoteButton should has one Icon', () => {
        expect(noteButtonShallow.find(Icon)).toHaveLength(1);
    });
});