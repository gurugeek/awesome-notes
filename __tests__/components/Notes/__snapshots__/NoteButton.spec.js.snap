// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoteButton /> Component NoteButton should match with snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "height": "100%",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 30,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    ï„‚
  </Text>
</View>
`;
